sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_task_adherence/task_adherence.prompty

```diff
@@ -21,97 +21,385 @@ inputs:
+Output a JSON object with the below keys in the given sequence:
+    1) explanation: a concise 15–60 word summary of how well the assistant followed the task
+    2) agent_perceived_task:  what the assistant believed it had to do, based on its response;
+    3) actual_task: what the task truly was, considering both system prompt and user latest query, with system taking precedence.
+    4) tools_used: List of tools used by the agent
+    5) expected_tool_calls: List of tools the agent should have used (if applicable)
+    6) score: an integer score from 1 (very poor) to 5 (excellent) on how well the agent resolved the user's intent within constraints
+
+SCORING RUBRIC WITH EXAMPLES
+============================
+
+**Score 5 - Fully Adherent:**
+- Executed intended task completely
+- Followed all mandatory system rules
+- Used tools correctly when required by system or task necessity
+- User's need fully satisfied within constraints
+
+Example A (single-turn)
+=======================
+CONVERSATION_HISTORY:
+User turn 1:
+  Suggest a good mystery book.
+
+AGENT_RESPONSE:
+  [TOOL_CALL] get_book(genre="mystery")
+  [TOOL_RESULT] "Gone Girl" by Gillian Flynn
+  I recommend *Gone Girl* by Gillian Flynn — a gripping psychological thriller.
+
+TOOL_DEFINITIONS:
+- get_book: Returns a book recommendation based on genre. (inputs: genre)
+
+EXPECTED OUTPUT:
+{
+  "explanation": "The assistant correctly identified the task, used the appropriate tool as needed, and provided a clear and suitable book recommendation.",
+  "agent_perceived_task": "recommend a mystery book",
+  "actual_task": "Suggest a good mystery book",
+  "tools_used": ["get_book"],
+  "expected_tool_calls": ["get_book"],
+  "score": 5
+}
+
+Example B (multi-turn)
+======================
+CONVERSATION_HISTORY:
+SYSTEM_PROMPT:
+Do not use tools unless explicitly requested by the user.
+
+User turn 1:
+  Can you recommend a good book?
+
+Agent turn 1:
+  Any preferred genre?
+
+User turn 2:
+  Historical fiction.
+
+AGENT_RESPONSE:
+  You might enjoy *The Book Thief* by Markus Zusak — a powerful story set during World War II.
+
+TOOL_DEFINITIONS:
+- get_book: Returns book suggestions based on genre. (inputs: genre)
+
+EXPECTED OUTPUT:
+{
+  "explanation": "The assistant correctly followed the system instruction by not using tools since the user never explicitly requested one. It engaged with the user to clarify preferences and gave a relevant recommendation.",
+  "agent_perceived_task": "recommend a historical fiction book without tool use",
+  "actual_task": "Recommend a historical fiction book (no tool allowed unless explicitly requested)",
+  "tools_used": [],
+  "expected_tool_calls": [],
+  "score": 5
+}
+
+**Score 4 - Mostly Adherent:**
+- Executed core intended task
+- Followed mandatory system rules
+- Minor execution gaps that don't prevent task completion
+- Tool usage mostly appropriate
+
+Example A (single-turn)
+=======================
+CONVERSATION_HISTORY:
+SYSTEM_PROMPT:
```

Comment:
@ghyadav : as per @JoseCSantos 's comment above, do we have System Prompt for Single Turn? BTW I am curious why it should be different for single turn vs multi-turn